### PR TITLE
Upgrade xunit to 2.3.0-beta4

### PR DESCRIPTION
**Pull request by natemcmaster**
Also, expands the list of xunit packages to include commonly used sub-packages, such as xunit.core.

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -19,8 +19,11 @@
     <ExternalDependency Include="Microsoft.NET.Test.Sdk" Version="15.3.0" Source="$(DefaultNuGetFeed)"/>
     <ExternalDependency Include="Moq" Version="4.7.49" Source="$(DefaultNuGetFeed)"/>
     <ExternalDependency Include="Newtonsoft.Json" Version="10.0.1" Source="$(DefaultNuGetFeed)"/>
-    <ExternalDependency Include="xunit.runner.visualstudio" Version="2.3.0-beta3-build3705" Source="$(DefaultNuGetFeed)"/>
-    <ExternalDependency Include="xunit" Version="2.3.0-beta3-build3705" Source="$(DefaultNuGetFeed)"/>
+    <ExternalDependency Include="xunit.analyzers" Version="0.6.1" Source="$(DefaultNuGetFeed)"/>
+    <ExternalDependency Include="xunit.assert" Version="2.3.0-beta4-build3742" Source="$(DefaultNuGetFeed)"/>
```
Comment (mikeharder) [inline build/dependencies.props]: Why explicitly specify `xunit.analyzers` and `xunit.assert`?

Comment (mikeharder) [inline build/dependencies.props]: I meant `xunit.assert` and `xunit.core`.  I see you are explicitly specifying a newer version of `xunit.assert`.

Comment (natemcmaster) [inline build/dependencies.props]: xunit is a metapackage of xunit.{assert, core, analyzers}. In some repros, we may want to reference only some of these but not all (e.g. some people may want to opt-out of analyzers).

Comment (mikeharder) [inline build/dependencies.props]: OK I get it now, this is just a list of versions, and each project chooses which subset they want.

**Pull request merged**